designs/Transitive-Dependencies.md

```diff
@@ -0,0 +1,150 @@
+
+## Audience: Who are we building this for?
+
+We are building this for .NET users, both old and new, to be successful with understanding the types of NuGet dependencies in their project(s) and solution(s).
+
+## What: What does this look like in the product?
+
+![](../resources/TransitiveDependencies/TransitiveVSPMUI.png)
+
+## Context and Scope
+
+-	https://github.com/NuGet/Home/issues/5887 
+-	https://developercommunity.visualstudio.com/content/idea/538119/add-dependency-tree-view-for-packagereferences.html 
+
+## Minimal Requirements
+
+-	Visual list of transitive dependencies.
+-	Ability to see a transitive dependency package details.
+-	Ability to determine where a transitive dependency originated.
+-	Ability to promote a transitive dependency to a top-level dependency.
+
+## Scenarios and User Experience
+
+### CLI
+
+We should consider a minor tweak to the existing experience of “dotnet list package --include-transitive” to provide the user with a sense of where the package came from. This could be a new column next to “Resolved” which says “Transitively Referenced” or “Referenced” and has a list of the top-level packages that requested the dependency.
+
+![](../resources/TransitiveDependencies/TransitiveDotNetCLI.png)
+
+### Visual Studio
+
+Within Visual Studio, there will be a new panel within the Visual Studio Package Manager UI named “Transitive Packages” in which all transitive packages will be displayed to the user. For the sake of not confusing the user, there will additionally be a header titled “Top-level Packages” where the current experience is today for top-level packages.
+
+**By default, a user’s transitive packages will be collapsed** as a project can have hundreds, if not thousands of transitive packages & would potentially impact performance displaying these many records on each load.
+
+When a user highlights a transitive package, they will see a pop-up that displays how the transitive dependency originated & what top-level package(s) are bringing it in.
+
+![](../resources/TransitiveDependencies/TransitiveVSPMUI.png)
+
+Finally, selecting a transitive dependency will provide the user the ability to “Install” the package as you normally would a top-level dependency. This will promote the transitive dependency to a top-level dependency, and the transitive dependency will now be placed in the top-level packages list instead.
+
+## Goals and Non-Goals
+
+### Goals
+
+-	View transitive dependency list displaying the version of each dependency.
+-	See what brought in a transitive dependency & what version. 
+-	Ability to promote a transitive dependency to a top-level dependency via typical NuGet PM UI commands.
+
+### Non-Goals
+
+-	Complete management of transitive dependencies.
+-	Reinvent common UX patterns used in Visual Studio today for package management actions.
+
+## Design
+
+### Elements
+
+There are three primary elements that will be added as part of this design to the existing Visual Studio Package Manager UI.
```

Comment:
The spec feels a bit more PM level, but now we're getting specific with XAML controls. Are these just high level concepts of what we think we're changing in the UI?
When you say ScrollViewer... you expect a new scrollbar for Top level and Transitive level to have their individual scrollbars?

Reply:
I have no concept of what controls should be used. I defer to the individual/implementation to use the best controls. These are just three elements I'm aware of to accomplish the design today.

Reply:
Ok, then I would think we shouldn't list them in the spec if they are being deferred.

Reply:
Added a sentence to help distinguish!